src/cleanfigure.m

```diff
@@ -939,6 +939,30 @@ function removeNaNs(meta, handle)
     topRight    = [xLim(2); yLim(2)];
 end
 % =========================================================================
+function data = limitPrecision(data)
+  % Limit the precision of the given data
+  
+  % Only do something if the data is not empty
+  if isempty(data) || all(~isfinite(data))
+      return
```

Comment:
Both cases that you describe are indeed problematic, but each of those lead to `data(isfinite(data))` to become empty and hence `range` to be empty and hence `data/[]` to fail.

So while there are two conditions you need at the beginning, there is only a single one that you need on `maxValue`. Moreover, currently it's impossible for `maxValue` to become `NaN` or `Inf` since those non-finite values don't get passed to the call to `max`.